homeassistant/components/iotawatt/sensor.py

```diff
@@ -161,20 +173,22 @@ def __init__(
         self._key = key
         data = self._sensor_data
         if data.getType() == "Input":
-            self._attr_unique_id = (
-                f"{data.hub_mac_address}-input-{data.getChannel()}-{data.getUnit()}"
-            )
+            self._attr_unique_id = f"{data.hub_mac_address}-input-{data.getChannel()}-{data.getUnit()}{self._name_suffix}"
         self.entity_description = entity_description
 
     @property
     def _sensor_data(self) -> Sensor:
         """Return sensor data."""
         return self.coordinator.data["sensors"][self._key]
 
+    @property
+    def _name_suffix(self) -> str:
+        return ""
+
     @property
     def name(self) -> str | None:
         """Return name of the entity."""
-        return self._sensor_data.getName()
+        return f"{self._sensor_data.getSourceName()}{self._name_suffix}"
```

Comment:
This is intended to be human readable. Ideally its append ` Accumulated` instead of `.accumulated`

Reply:
Is it a bit cleaner to leave this as-is and override the name property in the `IotaWattAccumulatingSensor`?

Reply:
I guess so. then I don't need suffix

Reply:
We actually still need a _name_suffix method as it's used in the unique_id part